joern-cli/frontends/jimple2cpg/src/main/scala/io/joern/jimple2cpg/util/ProgramHandlingUtil.scala

```diff
@@ -1,158 +1,136 @@
-              new File(dirName).mkdirs()
-              try {
-                if (destFile.exists()) destFile.delete()
-                Using.resource(zip.getInputStream(entry)) { input =>
-                  Files.copy(input, destFile.toPath)
-                }
-                destFile.deleteOnExit()
-                Option(destFile.getAbsolutePath)
-              } catch {
-                case e: Exception =>
-                  logger
-                    .warn(
-                      s"Encountered an error while extracting entry ${entry.getName} from archive ${zip.getName}.",
-                      e
-                    )
-                  Option.empty
-              }
-            })
-            .toList
-        )
+  object ClassFile {
+    def getPackagePathFromByteCode(fis: FileInputStream): Option[String] = {
+      val cr                   = new ClassReader(fis)
+      var path: Option[String] = None
+      sealed class ClassNameVisitor extends ClassVisitor(Opcodes.ASM9) {
+        override def visit(
+          version: Int,
+          access: Int,
+          name: String,
+          signature: String,
+          superName: String,
+          interfaces: Array[String]
+        ): Unit = {
+          path = Some(name)
+        }
       }
-    } catch {
-      case e: Exception => throw new RuntimeException(s"Error extracting files from archive at ${zf.getName}", e)
-    } finally {
-      FileUtils.deleteDirectory(zipTempDir.toFile)
+      val rootVisitor = new ClassNameVisitor()
+      cr.accept(rootVisitor, SKIP_CODE)
+      path
     }
-  }
 
-  /** Retrieve parseable files from archive types.
-    */
-  def extractSourceFilesFromArchive(sourceCodeDir: String, archiveFileExtensions: Set[String]): List[String] = {
-    val archives =
-      if (
-        new File(sourceCodeDir).isFile &&
-        archiveFileExtensions.map(sourceCodeDir.endsWith).reduce((a, b) => a && b)
-      ) {
-        List(sourceCodeDir)
-      } else {
-        SourceFiles.determine(sourceCodeDir, archiveFileExtensions)
-      }
-    archives.flatMap { x => unzipArchive(new ZipFile(x), sourceCodeDir) }
+    def getPackagePathFromByteCode(file: File): Option[String] =
+      Try(file.fileInputStream.apply(getPackagePathFromByteCode))
+        .recover {
+          case e: Throwable => {
+            logger.warn(s"Error reading class file ${file.canonicalPath}", e)
+            None
+          }
+        }
+        .getOrElse(None)
   }
+  sealed class ClassFile(val file: File, val packagePath: Option[String]) {
+    def this(file: File) = this(file, ClassFile.getPackagePathFromByteCode(file))
+
+    // Test that the path separator is always unix
+    val components: Option[Array[String]] = packagePath.map(_.split("/"))
 
-  /** Removes all files in the temporary unpacking directory.
-    */
-  def clean(): Unit = {
-    FileUtils.deleteDirectory(getUnpackingDir.toFile)
+    val fqcn: Option[String] = components.map(_.mkString("."))
```

Comment:
I've changed my mind about this `fqcn` thing, I think it'll be best as the full wording
```suggestion
    val fullyQualifiedClassName: Option[String] = components.map(_.mkString("."))
```